src/MDMSpringTimingCurveGenerator.m

```diff
@@ -0,0 +1,78 @@
+/*
+ Copyright 2017-present The Material Motion Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDMSpringTimingCurveGenerator.h"
+
+#import "MDMSpringTimingCurve.h"
+
+#import <UIKit/UIKit.h>
+
+@implementation MDMSpringTimingCurveGenerator
+
+- (instancetype)initWithDuration:(NSTimeInterval)duration dampingRatio:(CGFloat)dampingRatio {
+  return [self initWithDuration:duration dampingRatio:dampingRatio initialVelocity:0];
+}
+
+- (nonnull instancetype)initWithDuration:(NSTimeInterval)duration
+                            dampingRatio:(CGFloat)dampingRatio
+                         initialVelocity:(CGFloat)initialVelocity {
+  self = [super init];
+  if (self) {
+    _duration = duration;
+    _dampingRatio = dampingRatio;
+    _initialVelocity = initialVelocity;
+  }
+  return self;
+}
+
+#pragma mark - NSCopying
+
+- (id)copyWithZone:(NSZone *)zone {
+  return [[[self class] allocWithZone:zone] initWithDuration:self.duration
+                                                dampingRatio:self.dampingRatio
+                                             initialVelocity:self.initialVelocity];
```

Comment:
You may wish to access the ivars directly here to avoid subclasses overriding the returned values.  A subclass overriding this method can be sure to correct any values in the created copy.

Reply:
I'll address this in a follow-up CL by making all of the classes final.